pulsar-broker/src/main/java/org/apache/pulsar/broker/TransactionMetadataStoreService.java

```diff
@@ -98,6 +100,7 @@ public TransactionMetadataStoreService(TransactionMetadataStoreProvider transact
         this.transactionOpRetryTimer = timer;
         this.tcLoadSemaphores = new ConcurrentLongHashMap<>();
         this.pendingConnectRequests = new ConcurrentLongHashMap<>();
+        this.internalPinnedExecutor = pulsarService.getTransactionExecutorProvider().getExecutor(this);
```

Comment:
The TransactionExecutorProvider also be used by Transaction Buffer Recovery, this will create risks for TC which is not able to get CPU cycles to process transactions of other topics, I think we should consider using a separate executor provider for TC, and for TB, make sure one topic using the same thread.